### PR TITLE
[XPU] Serve Qwen3-Omni speech on Intel XPU, talker and Code2Wav with XPUGraphs

### PR DESCRIPTION
**Pull request by siju-samuel**
## Motivation

Support Qwen3-Omni speech — talker and Code2Wav — with XPUGraphs on Intel XPU.

Both speech-stage graph gates answered `False` on XPU, so the thinker could only
run `--text-only`. The talker graph is what puts the pipeline under realtime:
eager decode runs at RTF 1.45, slower than the audio it produces; the graph brings
that to 0.73.

## Modifications

1. **`enable_code2wav_graph()` and `enable_talker_graph()`: `False` → `True`** in
   `XPUOmniPlatform`.
2. **`SGLangModelRunner.init_cuda_graphs` pins SDPA around the capture on XPU.**
   The talker's capture died at startup with *"Graph nodes cannot depend on events
   from outside the graph"* — XPU's default SDPA dispatch records an event the graph
   did not create. The engines reach SDPA through model code SGLang's capture does
   not wrap, so the choice is made around the whole capture.
3. **`code2wav.gpu_memory_fraction: 0.02` → `0.05`** in the XPU profile. The
   fraction is of the whole card: 2.8 GiB on a 141 GiB H200, but 490 MiB on a
   24 GB B60, which left less graph budget than the capture keys need.

The Code2Wav runner also stops naming `torch.cuda` directly and asks the platform
for the graph backend #1911 landed. New config:
`examples/configs/qwen3_omni_speech_xpu_b60.yaml` (thinker TP=8, talker on card 6,
code2wav on card 7).

10 files, +702/−127.

**Non-XPU platforms are untouched, structurally.** Both capture pins — the SDPA
backend choice and transformers' packed-sequence probe — sit behind an `is_xpu()`
gate, and `init_cuda_graphs` enters nothing off XPU. CUDA, ROCm, MUSA and NPU reach
the same capture, with the same interpreter state, as before this PR. Asserted by
tests parametrised over `is_xpu`.

## Related Issues

- Tracking issue #1588.
- Builds on #1679 (XPU decode-graph backend opt-in).
- Depends on #1911 (merged), which landed `sglang_omni/platforms/device_graph.py`;
  this PR consumes its `DeviceGraphBackend` rather than keeping a private
  graph-type lookup.

## Accuracy Test

| | result |
|---|---|
| Round-trip WER, Qwen3-ASR judge | **0.0%** |
| Code2Wav graph replay vs eager | **bit-exact** at all four keys |

Replay is compared against eager under the same SDPA pin. Against *unpinned* eager
it would measure which attention kernel ran, not whether capture recorded
faithfully — on XPU the SDPA backends differ from each other in bf16 at these
shapes, and the default dispatch switches backend with the frame count.

## Benchmark & Profiling

8× Intel Arc Pro B60, weights warm. Talker decode graph A/B, two arms with only
`talker_ar.engine.disable_cuda_graph` changing and Code2Wav publishing 4/4 keys in
both, so the delta is the talker alone:

| run | graph (min) | eager (min) | speedup | RTF |
|---|---|---|---|---|
| quiet host | 1.88 s | 3.73 s | **1.98×** | 0.73 vs 1.45 |
| contended host | 2.01 s | 4.29 s | **2.14×** | 0.82 vs 1.58 |

Per-arm minimum, 4 warm requests per arm. The minima agree within ~8% across runs
while medians swing 1.65–2.02× with host load, so the floor is the statistic that
survives a shared machine. n=4 is thin; a CI-gating number wants the 5-run protocol
on a quiet host.

Code2Wav's own step goes 23.4–25.4 ms → 18.5–18.7 ms (1.24–1.37× on that step).
That is ~20 ms of a ~1.9 s request and the stage overlaps talker generation, so it
buys per-chunk jitter rather than throughput.

**Reproduce:**

```bash
sgl-omni serve --config examples/configs/qwen3_omni_speech_xpu_b60.yaml \
    --model-path <Qwen3-Omni-30B-A3B-Instruct> --port 8114

# eager arm
sgl-omni serve --config examples/configs/qwen3_omni_speech_xpu_b60.yaml \
    --model-path <model> --talker_ar.engine.disable_cuda_graph true
```

Confirm the graphs engaged rather than assuming:

```bash
grep "Capture target decode XPU graph end" <serve.log>   # talker
grep -o '"published_graph_count":[0-9]*' <serve.log>      # code2wav -> 4
```

## Checklist

- [x] Format your code according with pre-commit.
- [x] Add unit tests.
- [x] Update documentation / docstrings / example tutorials as needed.
- [x] Provide throughput / latency benchmark results and accuracy evaluation results as needed.
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.

## CI

`/tag-and-rerun-ci qwen3-tts qwen3-asr`

### CODE REVIEW

File: sglang_omni/models/qwen3_omni/components/code2wav_cuda_graph.py [hunk 1]
```diff
@@ -487,18 +594,18 @@ def _capture_graph(
...
-        self._cuda.warmup(
+        self._device_api.warmup(
             self._model,
             static_input,
             iterations=self._WARMUP_ITERATIONS,
             device=self._device,
             stream=stream,
         )
-        graph, static_output = self._cuda.capture(
+        graph, static_output = self._device_api.capture(
```
Comment (Ratish1) [inline sglang_omni/models/qwen3_omni/components/code2wav_cuda_graph.py]: the graph is recorded under `_xpu_capture_pins()`, but the eager reference immediately below runs after that context closes. on xpu it can select a different sdpa/mask path, so `torch.equal()` can reject a valid capture and disable code2wav graphs. keep the pins around the eager reference and equivalence replay.

File: sglang_omni/platforms/xpu.py [hunk 2]
```diff
@@ -65,6 +66,17 @@ def get_decode_cuda_graph_backend(self) -> str | None:
...
         return Backend.FULL
 
+    def get_graph_capture_sdpa_backends(self) -> tuple["SDPBackend", ...]:
+        # The default selection is not capturable: it records an event the graph
+        # did not create. Naming any backend avoids that, so name them in order.
+        from torch.nn.attention import SDPBackend
+
+        return (
+            SDPBackend.FLASH_ATTENTION,
+            SDPBackend.EFFICIENT_ATTENTION,
```
Comment (Ratish1) [inline sglang_omni/platforms/xpu.py]: `sdpa_kernel()` treats this tuple as an enabled set because `set_priority` is false, and pinned torch xpu reaches math before its unsupported efficient branch. `EFFICIENT_ATTENTION` therefore cannot be selected by this policy. can we describe this as the enabled capture-safe set, keep flash + math, and remove the ordering/efficient assertions?

File: sglang_omni/platforms/interface.py [hunk 3]
```diff
@@ -91,3 +93,22 @@ def get_decode_cuda_graph_backend(self) -> str | None:
...
+
+    def get_graph_capture_sdpa_backends(self) -> tuple["SDPBackend", ...]:
+        """SDPA backends to pin while capturing a graph, in preference order.
+
+        Empty leaves dispatch alone: pinning on a platform that does not need it
+        would only narrow what the capture may choose from.
+        """
+        return ()
+
+    def graph_capture_attention(self) -> AbstractContextManager[None]:
```
Comment (Ratish1) [inline sglang_omni/platforms/interface.py]: nit: `sdpa_kernel()` yields `{}`, while the empty path yields `None`, so this return type should be `AbstractContextManager[object]` rather than `AbstractContextManager[None]`.

**Review by Ratish1 — APPROVED** [review]
LGTM
